pkg/logpoller/service.go

```diff
@@ -0,0 +1,217 @@
+package logpoller
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/xssnick/tonutils-go/address"
+	"github.com/xssnick/tonutils-go/ton"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/logger"
+	"github.com/smartcontractkit/chainlink-common/pkg/services"
+
+	"github.com/smartcontractkit/chainlink-ton/pkg/logpoller/types"
+)
+
+// TON LogPoller Service
+//
+// This package implements a log polling service for TON blockchain.
+// It monitors external message outputs from specified addresses and
+// applies filtering logic to support cross-chain message detection.
+
+// service is the main TON log polling service implementation.
+// It continuously polls the TON masterchain, discovers new blocks, and processes
+// external messages from registered filter addresses.
+type service struct {
+	services.Service
+	eng *services.Engine // Service engine for lifecycle management
+
+	lggr    logger.SugaredLogger // Logger instance
+	client  ton.APIClientWrapped // TON blockchain client
+	filters FilterStore          // Registry of active filters
+	loader  TxLoader             // Transaction loader returning loaded txs
+	indexer TxIndexer            // Transaction indexer returning indexed logs
+	store   LogStore             // Log storage (MVP: in-memory, to be replaced with ORM)
+
+	pollPeriod         time.Duration // How often to poll for new blocks
+	lastProcessedBlock uint32        // Last processed masterchain sequence number
+}
+
+// NewService creates a new TON log polling service instance
+func NewService(lggr logger.Logger, opts *ServiceOptions) Service {
```

Comment:
Weird that this `ServiceOptions` type is in a different file and not here where it's used